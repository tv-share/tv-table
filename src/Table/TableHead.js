import React, { Component } from "react";
import PropTypes from "prop-types";

class TableHead extends Component {

	renderItems() {
		const { items } = this.props;
		return items.map(({ label, id, onClick }) => {
			return <th id={id} onClick={onClick} key={id}>{label}</th>;
		});
	}

	renderCheckBox() {
		const { withCheckBox, options: { onChange, checked } } = this.props;
		if(withCheckBox) {
			return (
				<th className="tv-checkbox">
					<input onChange={onChange} checked={checked} type="checkbox" />
				</th>
			);
		}
	}

	render() {
		const { withCheckBox, options: { className } } = this.props;
		return (
			<thead className={`tv-thead${className ? ` ${className}` : ""}`}>
				<tr>
					{this.renderCheckBox()}
					{this.renderItems()}
				</tr>
			</thead>
		);
	}
}

TableHead.defaultProps = {
	items: [],
	options: {}
};

TableHead.propTypes = {
	items: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		label: PropTypes.string,
		onClick: PropTypes.func
	})).isRequired,
	options: PropTypes.shape({
		className: PropTypes.string,
		onChange: PropTypes.func,
		checked: PropTypes.bool
	}),
	className: PropTypes.string,
	withCheckBox: PropTypes.bool
};

export default TableHead;