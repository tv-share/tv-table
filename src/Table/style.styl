table.tv-table
    font-family "Roboto", "arial", "sans-serif"
    width 100%
    border 1px solid #d0d0d0
    font-size 13px

    thead.tv-thead
        color #999
        letter-spacing 1px
        position relative
        border-bottom 1px solid #d0d0d0
        text-align left

        > tr > th
            padding 10px

            &.tv-checkbox
                width 1px


    tbody.tv-tbody
        position relative
        text-align left
        font-size 1.2em

        > tr > td
            padding 10px

            &.tv-checkbox
                width 1px