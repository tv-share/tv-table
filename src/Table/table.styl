table.tv-table
    font-family "Roboto", "arial", "sans-serif"
    width 100%
    border 1px solid #d0d0d0