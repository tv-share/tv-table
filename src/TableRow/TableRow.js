import React, { Component } from "react";
import PropTypes from "prop-types";

class TableRow extends Component {

	renderCheckBox() {
		if(this.props.withCheckBox) {
            if(!this.props.checkboxOptions) throw new Error("Parameter 'checkboxOptions' is required when 'withCheckBox' is true.");
            const { id, checkboxOptions: { selected, onChange }} = this.props;

			return (
				<td className="tv-checkbox">
					<input id={id} checked={selected} onChange={onChange} type="checkbox" />
				</td>
			);
		}
	}

	render() {
		const { id, data, withCheckBox, className, onClick } = this.props;
		return (
			<tr onClick={onClick} id={id} className={`tv-row${withCheckBox ? " -pointer" : ""}${className ? ` ${className}` : ""}`}>
				{this.renderCheckBox()}
				{
					data.map(d => {
						return <td key={d}>{d}</td>;
					})
				}
			</tr>
		);
	}
}

TableRow.defaultProps = {
	data: []
};

TableRow.propTypes = {
	id: PropTypes.string.isRequired,
	data: PropTypes.array.isRequired,
	className: PropTypes.string,
	onClick: PropTypes.func,
	withCheckBox: PropTypes.bool
};

export default TableRow;