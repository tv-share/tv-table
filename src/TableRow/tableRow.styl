table.tv-table tr.tv-row
    > td
        padding 10px
        font-size 1.1em
        border-top 1px solid #d0d0d0

        &.tv-checkbox
            width 1px

    &.-pointer
        cursor pointer

    &:hover
        background-color #f2f2f2