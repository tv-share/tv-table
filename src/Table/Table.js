import React, { Component } from 'react';

import TableHead from "../TableHead";

class Table extends Component {
    render() {
        const { head, withCheckBox } = this.props;
        return (
            <table className="tv-table">
                <TableHead withCheckBox={withCheckBox} items={head} />
            </table>
        );
    }
}

export default Table;