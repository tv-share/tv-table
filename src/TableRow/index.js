import "./tableRow.*";
import TableRow from "./TableRow";

export default TableRow;