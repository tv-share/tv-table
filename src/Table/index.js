import "./table.*";
import Table from "./Table";

export default Table;