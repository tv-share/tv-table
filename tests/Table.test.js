import React from 'react';
import renderer from 'react-test-renderer';

import Table from "../src/Table/Table";
import samples from "./Table.sample";
import TableRow from '../src/TableRow/TableRow';

console.log("samples", samples);

const fn = () => { };

const { headItems, bodyItems } = samples;

it('renders correctely - without checkbox', () => {
	const component = renderer.create(
		<Table headItems={headItems} >
            {
                bodyItems.map(({ id, name, tel }) => {
                    return <TableRow key={id} id={id} data={{ name, tel }} />;
                })
            }
        </Table>
	);

	expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctely - with checkbox', () => {
	const component = renderer.create(
		<Table withCheckBox headItems={headItems} >
            {
                bodyItems.map(({ id, name, tel }) => {
                    return <TableRow key={id} id={id} data={{ name, tel }} />;
                })
            }
        </Table>
	);

	expect(component.toJSON()).toMatchSnapshot();
});