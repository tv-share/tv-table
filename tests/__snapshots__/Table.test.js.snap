// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctely - with checkbox 1`] = `
<table
  className="tv-table"
>
  <thead
    className="tv-thead"
  >
    <tr>
      <th
        className="tv-checkbox"
      >
        <input
          checked={undefined}
          onChange={undefined}
          type="checkbox"
        />
      </th>
      <th
        id="name"
        onClick={undefined}
      >
        NOME
      </th>
      <th
        id="tel"
        onClick={undefined}
      >
        TELEFONE
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="tv-row -pointer"
      id="1"
      onClick={undefined}
    >
      <td
        className="tv-checkbox"
      >
        <input
          type="checkbox"
        />
      </td>
      <td>
        Thiago Loddi
      </td>
      <td>
        123
      </td>
    </tr>
    <tr
      className="tv-row -pointer"
      id="2"
      onClick={undefined}
    >
      <td
        className="tv-checkbox"
      >
        <input
          type="checkbox"
        />
      </td>
      <td>
        Andre Loddi
      </td>
      <td>
        456
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders correctely - without checkbox 1`] = `
<table
  className="tv-table"
>
  <thead
    className="tv-thead"
  >
    <tr>
      <th
        id="name"
        onClick={undefined}
      >
        NOME
      </th>
      <th
        id="tel"
        onClick={undefined}
      >
        TELEFONE
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="tv-row"
      id="1"
      onClick={undefined}
    >
      <td>
        Thiago Loddi
      </td>
      <td>
        123
      </td>
    </tr>
    <tr
      className="tv-row"
      id="2"
      onClick={undefined}
    >
      <td>
        Andre Loddi
      </td>
      <td>
        456
      </td>
    </tr>
  </tbody>
</table>
`;
